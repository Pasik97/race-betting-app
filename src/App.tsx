import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter, Route, Switch } from "react-router-dom";
import styled from 'styled-components';
import TopBar from 'modules/TopBar/TopBar';
import HomePage from 'modules/HomePage/HomePage';
import RacesPage from 'modules/RacesPage/RacesPage';
import configureStore from 'store/configureStore';
import media from 'utils/media';

const AppWrapper = styled.div`
   margin: 0 8px;

   ${media.fromTablet} {
      margin: 0 128px;
   }
   
   ${media.fromDesktop} {
      margin: 0 25%;
   }
`;

const store = configureStore();

const App: React.FC = () => (
   <Provider store={store}>
      <BrowserRouter>
         <TopBar />
         <AppWrapper>
            <Switch>
               <Route exact path='/' component={HomePage} />
               <Route exact path='/races' component={RacesPage} />
            </Switch>
         </AppWrapper>
      </BrowserRouter>
   </Provider>
);

export default App;
