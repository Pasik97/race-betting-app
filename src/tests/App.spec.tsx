import { create } from 'react-test-renderer';
import App from '../App';

describe('<App />', () => {
   it('should render properly', () => {
      const component = create(<App />);

      expect(component.toJSON()).toMatchSnapshot();
   });
});
