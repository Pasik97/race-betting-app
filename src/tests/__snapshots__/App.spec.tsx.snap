// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render properly 1`] = `
Array [
  <header
    className="sc-bdvvtL dktecm"
  >
    <a
      className="sc-gsDKAQ gtcDQK"
      href="/"
      onClick={[Function]}
      title="Race betting system"
    >
      Race betting system
    </a>
    <div
      className="sc-dkPtRN jUPLyL"
    >
      <a
        className="sc-hKwDye caybUZ"
        href="/races"
        onClick={[Function]}
      >
        Races
      </a>
    </div>
  </header>,
  <div
    className="sc-cxpSdN fiqEb"
  >
    <p
      className="sc-eCImPb eNWUA"
    >
      Welcome in races betting application. Use navigation to see current races.
    </p>
  </div>,
]
`;
